sql/analyzer/replace_window_names.go

```diff
@@ -0,0 +1,183 @@
+			newExprs := make([]sql.Expression, len(window.SelectExprs))
+			for i, expr := range window.SelectExprs {
+				newExprs[i], err = expression.TransformUp(expr, func(e sql.Expression) (sql.Expression, error) {
+					uf, ok := e.(*expression.UnresolvedFunction)
+					if !ok {
+						return e, nil
+					}
+					if uf.Window == nil {
+						return e, nil
+					}
+					newWindow, err := resolveWindowDef(uf.Window, wn.WindowDefs)
+					if err != nil {
+						return nil, err
+					}
+					return uf.WithWindow(newWindow), nil
+				})
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			return plan.NewWindow(newExprs, window.Child), nil
+		}
+		return n, nil
+	})
+}
+
+// checkCircularWindowDef verifies that window references terminate
+// with concrete definitions. We use a linked-list algorithm
+// because a sql.WindowDefinition can have at most one [Ref].
+func checkCircularWindowDef(windowDefs map[string]*sql.WindowDefinition) error {
+	var head, tail *sql.WindowDefinition
+	for _, def := range windowDefs {
+		if def.Ref == "" {
+			continue
+		}
+		head = def
+		head = windowDefs[head.Ref]
+		tail = def
+		for head != nil && tail != nil && head != tail {
+			tail = windowDefs[tail.Ref]
+			head = windowDefs[head.Ref]
+			if head != nil {
+				head = windowDefs[head.Ref]
+			}
+		}
+		if head != nil && head == tail {
+			return sql.ErrCircularWindowInheritance.New()
+		}
+	}
+	return nil
+}
+
+// resolveWindowDef uses DFS to walk the [windowDefs] adjacency list, resolving and merging
+// all named windows required to define the topmost window of concern.
+// A WindowDefinition is considered resolved when its [Ref] is empty. Otherwise, we recurse
+// to define that Ref'd window, before finally merging the resolved ref with the original window
+// definition.
+// A sql.WindowDef can have at most one named reference.
+// We cache merged definitions in [windowDefs] to aid subsequent lookups.
+func resolveWindowDef(n *sql.WindowDefinition, windowDefs map[string]*sql.WindowDefinition) (*sql.WindowDefinition, error) {
+	// base case
+	if n.Ref == "" {
+		return n, nil
+	}
+
+	var err error
+	ref, ok := windowDefs[n.Ref]
+	if !ok {
+		return nil, sql.ErrUnknownWindowName.New(n.Ref)
+	}
+
+	// recursively resolve [n.Ref]
+	ref, err = resolveWindowDef(ref, windowDefs)
+	if err != nil {
+		return nil, err
+	}
+
+	// [n] is fully defined by its attributes merging with the named reference
+	n, err = mergeWindowDefs(n, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	if n.Name != "" {
+		// cache lookup
+		windowDefs[n.Name] = n
+	}
+	return n, nil
+}
+
+// mergeWindowDefs combines the attributes of two window definitions or returns
+// an error if the two are incompatible. [def] should have a reference to
+// [ref] through [def.Ref], and the return value drops the reference to indicate
+// the two were properly combined.
+func mergeWindowDefs(def, ref *sql.WindowDefinition) (*sql.WindowDefinition, error) {
```

Comment:
I can't think of any reason why we can't do it the way you want, but i just did what MySQL does:
```
ERROR 3581 (HY000): A window which depends on another cannot define partitioning.
ERROR 3582 (HY000): Window 'w1' has a frame definition, so cannot be referenced by another window.
ERROR 3583 (HY000): Window '<unnamed window>' cannot inherit 'w1' since both contain an ORDER BY clause.
```